netbox/data_source_netbox_virtual_machines.go

```diff
@@ -0,0 +1,232 @@
+	return &schema.Resource{
+		Read: dataSourceNetboxVirtualMachineRead,
+		Schema: map[string]*schema.Schema{
+			"search": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:     schema.TypeString,
+				},
+			},
+			"vms": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cluster_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"comments": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"config_context": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"custom_fields": {
+							Type:     schema.TypeMap,
+							Computed: true,
+						},
+						"disk_size_gb": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"local_context_data": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"memory_mb": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"platform_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"primary_ip": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"primary_ip4": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"primary_ip6": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"role_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"site_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tag_ids": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeInt,
+							},
+						},
+						"tenant_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"vcpus": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"vm_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceNetboxVirtualMachineRead(d *schema.ResourceData, m interface{}) error {
+	api := m.(*client.NetBoxAPI)
+
+	params := virtualization.NewVirtualizationVirtualMachinesListParams()
+
+	if search, ok := d.GetOk("search"); ok {
+		var searchParams = search.(map[string]interface{})
+		for k, v := range searchParams {
+			switch k {
+			case "cluster_id":
+				var clusterString = v.(string)
+				params.ClusterID = &clusterString
+			case "cluster_group":
+				var clusterGroupString = v.(string)
+				params.ClusterGroup = &clusterGroupString
+			case "name":
+				var nameString = v.(string)
+				params.Name = &nameString
+			case "region":
+				var siteString = v.(string)
+				params.Site = &siteString
```

Comment:
This should probably say regionString etc.

Reply:
Oops. Fixed